kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/camelRoute/CamelRouteParseCatalogTest.java

```diff
@@ -0,0 +1,178 @@
+package io.kaoto.backend.metadata.parser.step.camelRoute;
+
+import io.kaoto.backend.metadata.ParseCatalog;
+import io.kaoto.backend.metadata.catalog.InMemoryCatalog;
+import io.kaoto.backend.model.parameter.BooleanParameter;
+import io.kaoto.backend.model.parameter.ObjectParameter;
+import io.kaoto.backend.model.parameter.Parameter;
+import io.kaoto.backend.model.parameter.StringParameter;
+import io.kaoto.backend.model.step.Step;
+import io.quarkus.test.junit.QuarkusTest;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertIterableEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@QuarkusTest
+public class CamelRouteParseCatalogTest {
+
+    @Inject
+    public void setParseCatalog(final CamelRouteParseCatalog parseCatalog) {
+        this.parseCatalog = parseCatalog;
+    }
+    private CamelRouteParseCatalog parseCatalog;
+
+    @Disabled("Git is down")
+    @Test
+    void shouldLoadFromGit() {
+        String camelGit = "https://github.com/apache/camel.git";
+        String tag = "camel-3.18.2";
+        InMemoryCatalog<Step> catalog = new InMemoryCatalog<>();
+
+        ParseCatalog<Step> camelParser =
+                parseCatalog.getParser(camelGit, tag);
+        List<Step> steps = camelParser.parse().join();
+
+        assertTrue(catalog.store(steps));
+
+        Step browseComponentSink = catalog.searchStepByID("browse-sink");
+        Step browseComponentSource = catalog.searchStepByID("browse-source");
+        Step browseComponentAction = catalog.searchStepByID("browse-action");
+
+        assertBrowseJsonHasBeenParsedCorrectly(
+                browseComponentSource, "source", false);
+        assertBrowseJsonHasBeenParsedCorrectly(
+                browseComponentSink, "sink", false);
+        assertBrowseJsonHasBeenParsedCorrectly(
+                browseComponentAction, "action", false);
+    }
+
+    @Test
+    void shouldLoadFromJar() {
+        String camelGit = "https://github.com/apache/camel/"
+                + "archive/refs/tags/camel-3.18.2.zip";
```

Comment:
For the test maybe better to have a fixed hardcoded url to prevent tests failing when we move to a newer version of camel on the properties. Wouldn't it?

Reply:
it depends what we want to check :-)
if in another test we check the content of the file pointed in the application.yaml that's great to keep an hardcoded one here.

Reply:
But I feel like that's not the point of Kaoto? We should parse any catalog source, so there shouldn't be any "hardcoded" one in the application.yml other than what the users want to put there. I think it doesn't make much sense, even for the "default" catalogs.

Reply:
There is one that is currently hardcoded in the application.yaml that i pointed in my first comment. I guess it is the one used by default. It seems very important that we check the one provided by default can be loaded. maybe it is covered by some other tests and this one can stay with a specific hardcoded version.

By experience in Eclipse Desktop Fuse Tooling, checking latest Camel catalog in tests was very helpful, we detected a lot of problems either in the catalog itself or API breakage or valid but not covered in our codebase cases on the various upgrade of catalog. We even had in plan to test against the snapshots build of the catalog (but never got prioritized)

Reply:
I guess we can open a issue to discuss this then. It shouldn't be a problem to always check the latest camel catalog (In fact, we could do that just using the git method and setting the tag to main, I think?), but I don't know if it is in the scope of this.

What do you think @Delawen

Reply:
I would create a separated PR for that. We are currently not checking the things on the `application.yaml` because we kind of assume that if we are able to parse the right format, it should work.

What we do on these tests is making sure that if the format is the right one, we are getting the expected step on the catalog.

Opened an issue to track this separatedly: https://github.com/KaotoIO/kaoto-backend/issues/155